core/actions/table.ts

```diff
@@ -1,247 +1,58 @@
-
-  /**
-   * Column(s) which may never be `NULL`.
-   *
-   * If set, the resulting assertion will fail if any row contains `NULL` values for these column(s).
-   */
-  nonNull?: string | string[];
-
-  /**
-   * General condition(s) which should hold true for all rows in the dataset.
-   *
-   * If set, the resulting assertion will fail if any row violates any of these condition(s).
-   */
-  rowConditions?: string[];
-}
-
-const ITableAssertionsProperties = () =>
-  strictKeysOf<ITableAssertions>()(["uniqueKey", "uniqueKeys", "nonNull", "rowConditions"]);
-
-/**
- * Configuration options for `dataset` actions, including `table`, `view` and `incremental` action types.
- */
-export interface ITableConfig
-  extends IActionConfig,
-    IDependenciesConfig,
-    IDocumentableConfig,
-    INamedConfig,
-    ITargetableConfig {
-  /**
-   * The type of the dataset. For more information on how this setting works, check out some of the [guides](guides)
-   * on publishing different types of datasets with Dataform.
-   */
-  type?: TableType;
-
-  /**
-   * Only allowed when the table type is `incremental`.
-   *
-   * If set to true, running this action will ignore the full-refresh option.
-   * This is useful for tables which are built from transient data, to ensure that historical data is never lost.
-   */
-  protected?: boolean;
-
-  /**
-   * BigQuery-specific warehouse options.
-   */
-  bigquery?: IBigQueryOptions;
-
-  /**
-   * Assertions to be run on the dataset.
-   *
-   * If configured, relevant assertions will automatically be created and run as a dependency of this dataset.
-   */
-  assertions?: ITableAssertions;
-
-  /**
-   * Unique keys for merge criteria for incremental tables.
-   *
-   * If configured, records with matching unique key(s) will be updated, rather than new rows being inserted.
-   */
-  uniqueKey?: string[];
-
-  /**
-   * Only valid when the table type is `view`.
-   *
-   * If set to true, will make the view materialized.
-   *
-   * For more information, read the [BigQuery materialized view docs](https://cloud.google.com/bigquery/docs/materialized-views-intro).
-   */
-  materialized?: boolean;
-}
-
-// TODO: This needs to be a method, I'm really not sure why, but it hits a runtime failure otherwise.
-export const ITableConfigProperties = () =>
-  strictKeysOf<ITableConfig>()([
-    "type",
-    "disabled",
-    "protected",
-    "name",
-    "bigquery",
-    "tags",
-    "uniqueKey",
-    "dependencies",
-    "hermetic",
-    "schema",
-    "assertions",
-    "database",
-    "columns",
-    "description",
-    "materialized",
-    "dependOnDependencyAssertions"
-  ]);
-
-/**
- * Context methods are available when evaluating contextable SQL code, such as
- * within SQLX files, or when using a [Contextable](#Contextable) argument with the JS API.
+ * @hidden
+ * This maintains backwards compatability with older versions.
+ * TODO(ekrekr): consider breaking backwards compatability of these in v4.
```

Comment:
for all the `TODO(erkerkr)` probably it is better to have `TODO(issue)`

Reply:
There are quite a lot of these, I'll make a triage plan and turn them into bugs in one go